### ISSUE TITLE
dolt can't commit changes without a database selected

### ISSUE THREAD

**Issue by zachmu** [dolthub/dolt#2658]
Can't reproduce this in a bats test, running it through bash seems to cause it to not trigger.

But in WSL (and in other windows contexts) the behavior is consistent: an empty dolt database.

```bash
timsehn@DESKTOP-K310NA1:/mnt/c/Users/timse/dolthub/git/pg2mysql/test$ cat out.sql
--
-- Generated by pg2mysql
--
set autocommit = off;
set foreign_key_checks = off;
DROP DATABASE IF EXISTS public;
CREATE DATABASE public;
CREATE TABLE public.test (
    `pk` integer NOT NULL,
    `c1` integer
);
INSERT INTO public.test VALUES (1, 1);
ALTER TABLE public.test
    ADD CONSTRAINT test_pkey PRIMARY KEY (pk);
timsehn@DESKTOP-K310NA1:/mnt/c/Users/timse/dolthub/git/pg2mysql/test$ dolt sql < out.sql
Rows inserted: 1 Rows updated: 0 Rows deleted: 0
timsehn@DESKTOP-K310NA1:/mnt/c/Users/timse/dolthub/git/pg2mysql/test$ ls
nickname.sql  out.sql  public  static_for_dolt.sql  test.sql
timsehn@DESKTOP-K310NA1:/mnt/c/Users/timse/dolthub/git/pg2mysql/test$ cd public/
timsehn@DESKTOP-K310NA1:/mnt/c/Users/timse/dolthub/git/pg2mysql/test/public$ dolt ls
No tables in working set
```

**Comment by timsehn** [comment]
This is bugged on Mac as well.

**Comment by timsehn** [comment]
```
test $ cat basic_test.mydump 
--
-- Generated by pg2mysql
--
set autocommit = off;
set foreign_key_checks = off;
DROP DATABASE IF EXISTS public;
CREATE DATABASE public;
CREATE TABLE public.basic_test (
    `pk` integer NOT NULL,
    `c1` integer
);
ALTER TABLE public.basic_test
    ADD CONSTRAINT basic_test_pkey PRIMARY KEY (pk);
test $ rm -rf public/
test $ dolt sql < basic_test.mydump 
Rows inserted: 0 Rows updated: 0 Rows deleted: 0
test $ cd public/
public $ dolt ls
No tables in working set

public $ dolt sql -q "show tables;"
+-------+
| Table |
+-------+
+-------+
```

**Comment by timsehn** [comment]
Skipped bats test in #2666

**Comment by timsehn** [comment]
This is resolved by #2685
